Utils/Services/ImageFetcher.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  ImageFetcher.swift
+//  Utils
+//
+//  Created by Nahuel Gladstein on 6/13/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import UIKit
+import ReactiveSwift
+
+public protocol ImageFetcherType {
+    
+    func fetchImage(_ imageURL: URL) -> SignalProducer<UIImage, ImageFetcherError>
+    
+}
+
+public enum ImageFetcherError: Error {
+    case invalidImageFormat
+    case fetchError(Error)
+}
+
+public class ImageFetcher: ImageFetcherType {
+    
+    /**
+     Fetches an image asynchronously from a URL, returning a value or an error in the SignalProducer when done.
+     - parameter imageURL: the url of the image to fetch.
+     */
+    public func fetchImage(_ imageURL: URL) -> SignalProducer<UIImage, ImageFetcherError> {
+        return URLSession.shared
```

Comment:
Okay. I would do the function receive a `URLSession` which may default to shared one, but we could give the user a `getCachedSession()` function if he wants to use a new cached session. I don't like caching "by hand", but if it is necessary the user will be able to do it; if it's not necessary to do it by hand, he could just use the cached session.